src/Statement/Rank.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+namespace Wikibase\DataModel\Statement;
+
+use InvalidArgumentException;
+
+/**
+ * @see https://www.mediawiki.org/wiki/Wikibase/DataModel#Ranks_of_Statements
+ * @see https://meta.wikimedia.org/wiki/Wikidata/Data_model_update#Ranks_and_order
+ *
+ * @since 4.4
+ *
+ * @licence GNU GPL v2+
+ * @author Thiemo Mättig
+ */
+class Rank {
+
+	/**
+	 * Higher values are more preferred.
+	 * TODO: Link to discussion/documentation that guarantees increasing order.
+	 */
+	const DEPRECATED = 0;
+	const NORMAL = 1;
+	const PREFERRED = 2;
+
+	/**
+	 * TODO: Add doc.
+	 */
+	const RANK_0 = self::DEPRECATED;
+	const RANK_1 = self::NORMAL;
+	const RANK_2 = self::PREFERRED;
```

Comment:
In those cases just use `0` instead of `Rank::RANK_0` as you hardcode the zero anyways.